### ISSUE TITLE
Version in conda-forge CI is 0.0.0

### ISSUE THREAD

**Issue by davidbrochart** [asphalt-framework/asphalt#73]
When installing Asphalt in the [conda-forge CI](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=678418&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d), we get:
```
Successfully installed asphalt-0.0.0
```
This prevents using `pip check` in [jupyverse's feedstock](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=692709&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d):
```
jupyverse-api 0.1.4 has requirement asphalt<5,>=4.11.0, but you have asphalt 0.0.0.
jupyverse-api 0.1.4 has requirement asphalt-web[fastapi]<2,>=1.1.0, but you have asphalt-web 0.0.0.
asphalt-web 0.0.0 has requirement asphalt~=4.8, but you have asphalt 0.0.0.
asphalt 0.0.0 has requirement asyncio-extras~=1.3, but you have asyncio-extras 0.0.0.
```
I cannot reproduce locally, but maybe there is an issue with `setuptools_scm`?

**Comment by agronholm** [comment]
I'm no expert in Anaconda matters, but it would appear that the build was done not from a git checkout, but an sdist?

**Comment by davidbrochart** [comment]
Yes, that's the preferred way of getting sources for Python packages on conda-forge.

**Comment by agronholm** [comment]
But setuptools_scm gets the version from git tags, so the sdist would not work properly. So either I should stop publishing sdists entirely, or find a mechanism in setuptools_scm that writes the version information in a file where setuptools could find it when installing from sdist?

**Comment by davidbrochart** [comment]
What about switching to [hatch](https://hatch.pypa.io)?

**Comment by agronholm** [comment]
I'm willing to entertain the idea, but this would probably require a PR because right now I have too much on my plate already with typeguard and anyio.
